packages/application-shell/src/components/application-shell/application-shell.spec.js

```diff
@@ -633,3 +639,596 @@ describe('when dispatching a loading notification', () => {
+    });
+    const rendered = renderApp(null, {
+      featureFlags: {
+        [PROJECT_EXTENSIONS]: false,
+      },
+      DEV_ONLY__loadNavbarMenuConfig: () => Promise.resolve([navbarMock]),
+    });
+    // Get the nav container, to narrow down the search area
+    const container = await rendered.findByTestId('left-navigation');
+    const navbarRendered = within(container);
+
+    const applicationLocale = operations.FetchLoggedInUser.me.language;
+    const mainMenuLabel = navbarMock.labelAllLocales.find(
+      localized => localized.locale === applicationLocale
+    );
+    await wait(() => {
+      expect(
+        navbarRendered.queryByText(mainMenuLabel.value)
+      ).not.toBeInTheDocument();
+    });
+  });
+});
+describe('when navbar menu items match given permissions', () => {
+  let operations;
+  beforeEach(() => {
+    operations = mocksForMc.createMockOperations({
+      FetchProject: {
+        project: ProjectMock.build({
+          allAppliedPermissions: [{ name: 'canManageOrders', value: true }],
+        }),
+      },
+    });
+    createGraphqlMockServer(xhrMock, {
+      operationsByTarget: { mc: operations },
+    });
+    selectProjectKeyFromUrl.mockReturnValue(
+      operations.FetchLoggedInUser.me.defaultProjectKey
+    );
+  });
+  it('should render item', async () => {
+    const navbarMock = ApplicationNavbarMenuMock.build({
+      permissions: ['ManageOrders'],
+    });
+    const rendered = renderApp(null, {
+      featureFlags: {
+        [PROJECT_EXTENSIONS]: false,
+      },
+      DEV_ONLY__loadNavbarMenuConfig: () => Promise.resolve([navbarMock]),
+    });
+    // Get the nav container, to narrow down the search area
+    const container = await rendered.findByTestId('left-navigation');
+    const navbarRendered = within(container);
+
+    const applicationLocale = operations.FetchLoggedInUser.me.language;
+    const mainMenuLabel = navbarMock.labelAllLocales.find(
+      localized => localized.locale === applicationLocale
+    );
+    await navbarRendered.findByText(mainMenuLabel.value);
+  });
+});
+describe('when navbar menu items do not match given permissions', () => {
+  let operations;
+  beforeEach(() => {
+    operations = mocksForMc.createMockOperations({
+      FetchProject: {
+        project: ProjectMock.build({
+          allAppliedPermissions: [{ name: 'canManageOrders', value: false }],
+        }),
+      },
+    });
+    createGraphqlMockServer(xhrMock, {
+      operationsByTarget: { mc: operations },
+    });
+    selectProjectKeyFromUrl.mockReturnValue(
+      operations.FetchLoggedInUser.me.defaultProjectKey
+    );
+  });
+  it('should not render item', async () => {
+    const navbarMock = ApplicationNavbarMenuMock.build({
+      permissions: ['ViewOrders'],
+    });
+    const rendered = renderApp(null, {
+      featureFlags: {
+        [PROJECT_EXTENSIONS]: false,
+      },
+      DEV_ONLY__loadNavbarMenuConfig: () => Promise.resolve([navbarMock]),
+    });
```

Comment:
Somehow, when reading this, I am starting to wonder if we not wastefully loose 10LoC on every it until it gets to something. Maybe this can be put into a `render` helper which just receives 1-2 args which is the variation?

Reply:
I though about this but decided to go with being more explicit and have things duplicated. I can have another look to see if some stuff can be simplified

Reply:
Yeah makes total sense. Always a hard balance how much to extract into test setup at the risk of having too much complexity there.